yente/data/delta.py

```diff
@@ -0,0 +1,108 @@
+from typing import Optional, TypedDict, Dict, List, Any
+from typing import AsyncGenerator, Tuple
+
+from yente.logs import get_logger
+from yente.data.dataset import Dataset
+from yente.data.loader import load_json_url, load_json_lines
+
+log = get_logger(__name__)
+
+
+class DeltaIndex(TypedDict):
+    versions: Dict[str, str]
+
+
+class EntityOp(TypedDict):
+    op: str
+    entity: Dict[str, Any]
+
+
+class DatasetLoader(object):
+    """A helper object for emitting entity operations to transition from one
+    loaded dataset version to the next."""
+
+    def __init__(
+        self, dataset: Dataset, base_version: Optional[str], force_full: bool = False
+    ) -> None:
+        self.dataset = dataset
+        self.target_version = dataset.version or "static"
+        self.base_version = base_version
+        self.force_full = force_full
+        self.delta_urls: Optional[List[Tuple[str, str]]] = None
+
+    @classmethod
+    async def build(
```

Comment:
What do you say to the loader taking a provider so that it can produce operations based on what we have in the delta files?

Reply:
Do you mean have the loader do what's in `yente.search.base:make_indexable` and the other iter_entity_docs function? Feels a bit like an overstepping of the data layer into backend concerns to me.

Reply:
You might be right there. We could also give a SearchProvider to `iter_entity_docs` so that it is able to take the JSON from the loader and turn it into something that can be passed to ES/OS as an operation.